core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -81,7 +89,38 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
     new File(subDir, filename)
   }
 
-  def getFile(blockId: BlockId): File = getFile(blockId.name)
+  /** Looks up a file by blockManagerId */
+  // This method should be kept in sync with
+  // org.apache.spark.network.shuffle.StandaloneShuffleBlockManager#getFile().
+  def getFile(filename: String,  blockManagerId: BlockManagerId): File = {
```

Comment:
should this be called `getLocalFile`? Also the javadoc should explicitly say that the given `blockManagerId` is expected to refer to an executor started on this particular machine (i.e. it should be local).